### ISSUE TITLE
Since video_player 1.0.0 was released, some versions do not build on Android.

### ISSUE THREAD

**Issue by hyejobs** [flutter/flutter#69754]
Since video_player 1.0.0 was released, 1.0.0, 0.11.1+4, 0.11.1+5 versions do not build on Android.
It only works fine on iOS. So I fixed version to 0.11.1+2.
Is there anything I need to check to use the latest version?

## Steps to Reproduce

1. Update video_player version to 0.11.1+4 or later.
2. flutter clean and pub get and build on Android/iOS device.

**Expected results:**
It should be successful on Android/iOS.

**Actual results:**
Build fails on Android and succeeds on iOS.

<details>
  <summary>Logs</summary>

```
[        ] /Users/user/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/video_player-1.0.0/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java:11: error: cannot find symbol
[   +1 ms] import io.flutter.FlutterInjector;
[        ]                  ^
[        ]   symbol:   class FlutterInjector
[        ]   location: package io.flutter
[        ] /Users/user/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/video_player-1.0.0/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java:77: error: cannot find symbol
[        ]     final FlutterInjector injector = FlutterInjector.instance();
[        ]           ^
[        ]   symbol:   class FlutterInjector
[  +10 ms]   location: class VideoPlayerPlugin
[        ] /Users/user/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/video_player-1.0.0/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java:77: error: cannot find symbol
[        ]     final FlutterInjector injector = FlutterInjector.instance();
[        ]                                      ^
[        ]   symbol:   variable FlutterInjector
[        ]   location: class VideoPlayerPlugin
[        ] 3 errors
[        ] FAILURE: Build failed with an exception.
[        ] * What went wrong:
[        ] Execution failed for task ':video_player:compileDebugJavaWithJavac'.
[        ] > Compilation failed; see the compiler error output for details.
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 22s
[ +381 ms] Running Gradle task 'assembleDebug'... (completed in 23.6s)
[   +5 ms] Exception: Gradle task assembleDebug failed with exit code 1
[   +2 ms] "flutter run" took 28,625ms.
[   +5 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:591:7)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:860:18)
           #3      _rootRunUnary (dart:async/zone.dart:1198:47)
           #4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #5      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #7      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #8      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
           #12     _rootRunUnary (dart:async/zone.dart:1198:47)
           #13     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #14     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #16     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #17     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #18     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #19     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #20     AndroidDevice.isLocalEmulator (package:flutter_tools/src/android/android_device.dart)
           #21     _rootRunUnary (dart:async/zone.dart:1198:47)
           #22     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #23     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #24     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #25     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #26     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #27     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #28     _rootRun (dart:async/zone.dart:1190:13)
           #29     _CustomZone.run (dart:async/zone.dart:1093:19)
           #30     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #31     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #32     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #33     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #34     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #35     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)

```

```
[✓] Flutter (Channel stable, 1.20.1, on Mac OS X 10.15.6 19G2021, locale ko-KR)
    • Flutter version 1.20.1 at /Users/user/Documents/flutter
    • Framework revision 2ae34518b8 (3 months ago), 2020-08-05 19:53:19 -0700
    • Engine revision c8e3b94853
    • Dart version 2.9.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Users/user/Downloads/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C505
    • CocoaPods version 1.9.1

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • Pixel 3a (mobile) • 94FAY0NHHD • android-arm64 • Android 10 (API 29)

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@hyejobs 
Tried `0.11.1+4`, `0.11.1+5` plugin versions on Android 8.1.0 device using latest stable (1.22.3) and was able to compile and run app without any issues.
I noticed that the latest `video_player` version `^1.0.0` is just released today (Nov 4) and there are no details in the [changelog](https://pub.dev/packages/video_player/changelog) yet, but I'd wait for couple of days for the team to update if there are any major changes that need to be done before using the plugin (but I don't think that is required). you should be able to run the plugin out of the box.

Also, I'd suggest you to upgrade to latest stable and then then run `flutter clean` after you upgrade your plugin version.

**Comment by qjatjr1108** [comment]
Hello

I have the same symptoms of failing the build the same.

```
11:22:06 android build command: flutter build apk lib/main_dev.dart
11:22:06 -----------------------------------------------------------------------------
11:22:06 
11:22:07 You are building a fat APK that includes binaries for android-arm, android-arm64, android-x64.
11:22:07 If you are deploying the app to the Play Store, it's recommended to use app bundles or split the APK to reduce the APK size.
11:22:07     To generate an app bundle, run:
11:22:07         flutter build appbundle --target-platform android-arm,android-arm64,android-x64
11:22:07         Learn more on: https://developer.android.com/guide/app-bundle
11:22:07     To split the APKs per ABI, run:
11:22:07         flutter build apk --target-platform android-arm,android-arm64,android-x64 --split-per-abi
11:22:07         Learn more on:  https://developer.android.com/studio/build/configure-apk-splits#configure-abi-split
11:22:07 Running Gradle task 'assembleRelease'...                        
11:23:55 Note: /Users/jenkins/flutter/.pub-cache/hosted/pub.dartlang.org/appsflyer_sdk-5.4.1+1/android/src/main/java/com/appsflyer/appsflyersdk/AppsflyerSdkPlugin.java uses unchecked or unsafe operations.
11:23:55 
11:23:55 Note: Recompile with -Xlint:unchecked for details.
11:23:59 
11:23:59 Note: /Users/jenkins/flutter/.pub-cache/hosted/pub.dartlang.org/multi_image_picker-4.7.14/android/src/main/java/com/vitanov/multiimagepicker/FileDirectory.java uses or overrides a deprecated API.
11:23:59 
11:23:59 Note: Recompile with -Xlint:deprecation for details.
11:24:01 
11:24:01 Note: Some input files use or override a deprecated API.
11:24:01 
11:24:01 Note: Recompile with -Xlint:deprecation for details.
11:24:01 
11:24:01 Note: Some input files use unchecked or unsafe operations.
11:24:01 
11:24:01 Note: Recompile with -Xlint:unchecked for details.
11:24:20 
11:24:20 Note: /Users/jenkins/flutter/.pub-cache/hosted/pub.dartlang.org/uni_links-0.4.0/android/src/main/java/name/avioli/unilinks/UniLinksPlugin.java uses or overrides a deprecated API.
11:24:20 
11:24:20 Note: Recompile with -Xlint:deprecation for details.
11:24:23 
11:24:23 /Users/jenkins/flutter/.pub-cache/hosted/pub.dartlang.org/video_player-0.11.1+5/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java:11: error: cannot find symbol
11:24:23 
11:24:23 import io.flutter.FlutterInjector;
11:24:23 
11:24:23                  ^
11:24:23 
11:24:23   symbol:   class FlutterInjector
11:24:23 
11:24:23   location: package io.flutter
11:24:23 
11:24:23 /Users/jenkins/flutter/.pub-cache/hosted/pub.dartlang.org/video_player-0.11.1+5/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java:77: error: cannot find symbol
11:24:23 
11:24:23     final FlutterInjector injector = FlutterInjector.instance();
11:24:23 
11:24:23           ^
11:24:23 
11:24:23   symbol:   class FlutterInjector
11:24:23 
11:24:23   location: class VideoPlayerPlugin
11:24:23 
11:24:23 /Users/jenkins/flutter/.pub-cache/hosted/pub.dartlang.org/video_player-0.11.1+5/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java:77: error: cannot find symbol
11:24:23 
11:24:23     final FlutterInjector injector = FlutterInjector.instance();
11:24:23 
11:24:23                                      ^
11:24:23 
11:24:23   symbol:   variable FlutterInjector
11:24:23 
11:24:23   location: class VideoPlayerPlugin
11:24:23 
11:24:23 3 errors
11:24:23 
11:24:23 
11:24:23 
11:24:23 FAILURE: Build failed with an exception.
11:24:23 
11:24:23 
11:24:23 
11:24:23 * What went wrong:
11:24:23 
11:24:23 Execution failed for task ':video_player:compileReleaseJavaWithJavac'.
11:24:23 
11:24:23 > Compilation failed; see the compiler error output for details.
```

The flutter version I'm using
```
flutter doctor -v
[✓] Flutter (Channel unknown, 1.20.4, on Mac OS X 10.15.7 19H2, locale ko-KR)
    • Flutter version 1.20.4 at /Users/teamblind_mac/flutter-sdk/flutter
    • Framework revision fba99f6cf9 (7 weeks ago), 2020-09-14 15:32:52 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/teamblind_mac/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[!] IntelliJ IDEA Ultimate Edition (version 2019.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 192.7761
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[!] IntelliJ IDEA Community Edition (version 2019.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    • Dart plugin version 192.7402
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins
```

Will it only work with the latest version of Flutter?

**Comment by plumhly** [comment]
when update flutter to v1.22.3, error gone!! thanks @darshankawar

**Comment by hyejobs** [comment]
When I got this error my flutter version was 1.20.1.
Solved after updating flutter to 1.22.3. thanks!

**Comment by darshankawar** [comment]
@hyejobs Glad to know your issue is resolved.
Closing this per your comment above.
